### PR TITLE
Update docs to account for always-on proxy injector and new `linkerd inject` behaviour

### LINKED ISSUE #288: improve autoinjector annotation docs
Much of the behavior of injection is documented in the CLI docs, but now that we have autoinjector on by default, we should have a version of this in the autoinject documentation in terms of annotations.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #288

### CODE REVIEW

File: linkerd.io/content/2/reference/proxy-configuration.md [hunk 1]
```diff
@@ -27,5 +27,4 @@ spec:
 ```
 
 Note that configuration overrides on proxies injected using the `linkerd inject`
-command is planned for release 2.4. Follow this
-[GitHub issue](https://github.com/linkerd/linkerd2/issues/2590/) for progress.
+command is supported as of versions stable-2.4.0 and edge-19.4.5.
```
Comment (ihcsim) [inline linkerd.io/content/2/reference/proxy-configuration.md]: I think when 2.4 lands, we can just remove this sentence; no needs to make a distinction between `linkerd inject` and auto-inject in this context.

File: linkerd.io/content/2/reference/cli/inject.md [hunk 2]
```diff
@@ -150,6 +181,9 @@ status: {}
 ---
 ```
 
+Note that for advanced use cases, you can have this output be produced directly
```
Comment (ihcsim) [inline linkerd.io/content/2/reference/cli/inject.md]: TIOLI: I think _advanced_ use cases can be a bit vague. We can just say if a user wants to see the injected YAML output, they can use `linkerd inject --manual`.

File: linkerd.io/layouts/shortcodes/cli/flags.html [hunk 3]
```diff
@@ -2,6 +2,12 @@
 {{ $opts := where .Options "Name" "!=" "help" }}
 {{ if ge (len $opts) 1 }}
 <h1>Flags</h1>
+{{ if eq .Name "inject" }}
+  Note that the flags that relate to proxy configuration are converted into
```
Comment (ihcsim) [inline linkerd.io/layouts/shortcodes/cli/flags.html]: IIUC, the request in #288 is to mention support of auto-inject annotations in the auto injection docs?

File: linkerd.io/content/2/reference/proxy-configuration.md (hunk 1 above)
Comment (alpeb) [reply linkerd.io/content/2/reference/proxy-configuration.md]: Agreed :+1:

File: linkerd.io/layouts/shortcodes/cli/flags.html (hunk 3 above)
Comment (alpeb) [reply linkerd.io/layouts/shortcodes/cli/flags.html]: Good point. I'll add a note in the `alpeb/autoinject-changes` branch.

Comment (alpeb) [reply linkerd.io/layouts/shortcodes/cli/flags.html]: (that's this branch.. I had a mix up with the other PR...)

**Comment by ihcsim** [comment]
Superseded by https://github.com/linkerd/website/pull/418.

**Pull request closed without merging**